src/opm/parser/eclipse/EclipseState/Grid/FieldProps.hpp

```diff
@@ -37,6 +38,177 @@ class Deck;
 class EclipseGrid;
 class TableManager;
 
+namespace keywords {
+
+template <typename T>
+struct keyword_info {
+    std::optional<std::string> unit = std::nullopt;
+    std::optional<T> scalar_init = std::nullopt;
+    bool multiplier = false;
+    bool top = false;
+    bool global = false;
+};
+
+
+
+inline bool isFipxxx(const std::string& keyword) {
+    // FIPxxxx can be any keyword, e.g. FIPREG or FIPXYZ that has the pattern "FIP.+"
+    // However, it can not be FIPOWG as that is an actual keyword.
+    if (keyword.size() < 4 || keyword == "FIPOWG") {
+        return false;
+    }
+    return keyword[0] == 'F' && keyword[1] == 'I' && keyword[2] == 'P';
+}
+
+
+namespace GRID {
+static const std::unordered_map<std::string, keyword_info<double>> double_keywords = {{"MULTPV",  keyword_info<double>{.scalar_init = 1.0}},
+                                                                                      {"NTG",     keyword_info<double>{.scalar_init = 1.0}},
```

Comment:
Designated initialisers are a C++20 feature.  Supporting this in C++17 mode appears to be a GCC extension.  Moreover, the extension is not available before GCC 8.  While really convenient, I don't think we can use this feature in OPM now.

Reply:
:angry: